Release v7.0.0 (#558)
bugsnag/bugsnag-unity: UPGRADING.md

Upgrading
=========

## 6.x to 7.x

When building using Unity 2019+, the Bugsnag SDK now uses a new method to intercept uncaught C# exceptions. This allows us access to the original exception object, meaning more accurate exception data and full support for inner exceptions.

This has slightly changed the format used for method names and signatures, which means that uncaught C# exceptions reported by `bugsnag-unity` V7+ will not group automatically in the Bugsnag dashboard with the equivalent exception reported by older versions of the library. 

All errors will still be reported, but you may wish to pay extra attention to new exceptions that come through once upgrading and group them accordingly.

> Handled errors and native crashes will continue grouping in the same way and are not affected by this upgrade.

Reviewer: ```suggestion
Handled errors and native crashes will maintain their existing groups and are not affected by this upgrade.
```

Upgrading
=========

## 6.x to 7.x

> ### The new Exception interceptor

Reviewer 2: I know `Exception` is a class name, but it irks me when I read it capitalised - in fact, we probably don't even need the h3 as it's the only noteworthy change.

> When building using Unity 2019+, the Bugsnag SDK now uses a new method to intercept uncaught C# exceptions. This allows us access to the original exception object, meaning more accurate exception data and full support for nested/inner exceptions.

Reviewer: I think "inner exception" is the correct C# term
```suggestion
When building using Unity 2019+, the Bugsnag SDK now uses a new method to intercept uncaught C# exceptions. This allows us access to the original exception object, meaning more accurate exception data and full support for inner exceptions.
```

> This means that uncaught C# exceptions previously reported will not group automatically in the Bugsnag dashboard with exceptions reported by the new notifier. 

Reviewer: ```suggestion
This means that uncaught C# exceptions reported by `bugsnag-unity` V7+ will not group automatically in the Bugsnag dashboard with the equivalent exception reported by older versions of the library. 
```

> This means that uncaught C# exceptions previously reported will not group automatically in the Bugsnag dashboard with exceptions reported by the new notifier. 

Reviewer 2: Should we add a bit more info here as it doesn't really say _why_ it won't group.

e.g.

> Due to differences in the way types were previously exposed in Unity's logging mechanism and the type information available when the exception is intercepted, errors will be reported subtly differently once your app is upgraded to this version.

Reviewer: Updated my suggestion.

> All errors will still be reported, but you may wish to pay extra attention to new exceptions that come through once upgrading and group them accordingly.

Reviewer: > group them accordingly.

What do you mean by this?  I don't think there's anything that can be done?

Author: Yeah this is not worded correctly, i suppose i thought we should mention that you can group by context or by hash, but perhaps it's not necessary to say that

Reviewer: ```suggestion
The impact of this will be that some errors will effectively be duplicated between older and newer versions of your app.
```

> Handled errors and native errors will remain grouped and are not effected by this upgrade.

Reviewer: ```suggestion
Handled errors and native crashes will continue grouping in the same way and are not affected by this upgrade.
```
